### PR TITLE
remove duplicated code in mesh deformation

### PR DESCRIPTION
**Pull request by tjhei**
The assembly+linear solve seems to be copy&pasted. We can get rid of that...

### CODE REVIEW

**Comment by tjhei** [comment]
Updated. I hope the code makes a bit more sense now with the comments added.

File: include/aspect/mesh_deformation/interface.h [hunk 1]
```diff
@@ -326,39 +326,31 @@ namespace aspect
...
-         * Set the boundary conditions for the solution of the elliptic
-         * problem, which computes the initial displacements of the internal
-         * vertices so that the mesh does not become too distorted due to
-         * motion of the surface. Displacements of vertices on the deforming
-         * surface are fixed according to the selected deformation plugins.
+         * Compute the initial constraints for the mesh displacement
+         * on the boundaries of the domain.  This is used on the mesh
+         * deformation boundaries to describe a displacement (initial
+         * topography) to be used during the simulation. The
+         * displacment is given by the active deformation plugins.
```
Comment (gassmoeller) [inline include/aspect/mesh_deformation/interface.h:333]:
```suggestion
         * displacement is given by the active deformation plugins.
```

File: include/aspect/mesh_deformation/interface.h [hunk 2]
```diff
@@ -326,39 +326,31 @@ namespace aspect
...
-         * motion of the surface. Velocities of vertices on the
-         * deforming surface are fixed according to the selected deformation
-         * plugins. Velocities of vertices on free-slip boundaries are
-         * constrained to be tangential to those boundaries. Velocities of
-         * vertices on no-slip boundaries are set to be zero. If a no-slip
-         * boundary is marked as additional tangential, then vertex velocities
-         * are constrained as tangential.
+         * Compute the constraints for the mesh velocity on the
+         * boundaries of the domain.  On the mesh deformation
+         * boundaries, the velocity is gien by the active deformation
```
Comment (gassmoeller) [inline include/aspect/mesh_deformation/interface.h:340]:
```suggestion
         * boundaries, the velocity is given by the active deformation
```

**Review by gassmoeller — COMMENTED** [review]
two small typos, otherwise good to go

**Comment by tjhei** [comment]
Thanks. Updated.

**Review by gassmoeller — APPROVED** [review]

**Pull request merged**